### PR TITLE
WIP - (BOLT-163) Configurable connection timeouts

### PR DESCRIPTION
**Pull request by grimradical**

### CODE REVIEW

File: lib/bolt/node/winrm.rb [hunk 1]
```diff
@@ -9,9 +9,14 @@ def initialize(host, port, user, password, shell: :powershell, **kwargs)
...
       @shell = shell
       @endpoint = "http://#{host}:#{port}/wsman"
-      @connection = ::WinRM::Connection.new(endpoint: @endpoint,
-                                            user: @user,
-                                            password: @password)
+      options = { endpoint: @endpoint,
+                  user: @user,
+                  password: @password }
+      if @timeout then
+        options[:receive_timeout] = @timeout
```
Comment (Iristyle) [inline lib/bolt/node/winrm.rb]: So far this is not working as expected... working to further investigate WinRM protocol and the gems behavior.

**Comment by grimradical** [comment]
Okay i think what I'm going to do with this is split things up...I believe I can get the basic connection timeouts (establishing an initial connection with ssh/winrm targets), but anything beyond that (in particular, winrm's command timeout) seems tricky/unreliable enough that I'm inclined to make a separate ticket for those. Stay tuned!

**Comment by Iristyle** [comment]
Yeah, separate effort around winrm sounds fine.  I'm still looking into the specifics there.

**Comment by Iristyle** [comment]
Cleaned up / working in #157

**Pull request closed without merging**